src/platform/input/game-pads.js

```diff
@@ -68,62 +168,763 @@ const MAPS = {
         ],
 
         axes: [
-            // Analogue Sticks
+            // Analog Sticks
             'PAD_L_STICK_X',
             'PAD_L_STICK_Y',
             'PAD_R_STICK_X',
             'PAD_R_STICK_Y'
-        ]
+        ],
+
+        mapping: 'standard'
+    },
+
+    DEFAULT_XR: {
+        buttons: [
+            // Back buttons
+            'XRPAD_TRIGGER',
+            'XRPAD_SQUEEZE',
+
+            // Axes buttons
+            'XRPAD_TOUCHPAD_BUTTON',
+            'XRPAD_STICK_BUTTON',
+
+            // Face buttons
+            'XRPAD_A',
+            'XRPAD_B'
+        ],
+
+        axes: [
+            // Analog Sticks
+            'XRPAD_TOUCHPAD_X',
+            'XRPAD_TOUCHPAD_Y',
+            'XRPAD_STICK_X',
+            'XRPAD_STICK_Y'
+        ],
+
+        mapping: 'xr-standard'
     }
 };
 
 const PRODUCT_CODES = {
     'Product: 0268': 'PS3'
 };
 
+const custom_maps = {};
+
+/**
+ * Retrieve the order for buttons and axes for given HTML5 Gamepad.
+ *
+ * @param {Gamepad} pad - The HTML5 Gamepad object.
+ * @returns {object} Object defining the order of buttons and axes for given HTML5 Gamepad.
+ */
+function getMap(pad) {
+    const custom = custom_maps[pad.id];
+    if (custom) {
+        return custom;
+    }
+
+    for (const code in PRODUCT_CODES) {
+        if (pad.id.indexOf(code) !== -1) {
+            const product = PRODUCT_CODES[code];
+
+            if (!pad.mapping) {
+                const raw = MAPS['RAW_' + product];
+
+                if (raw) {
+                    return raw;
+                }
+            }
+
+            return MAPS[product];
+        }
+    }
+
+    if (pad.mapping === 'xr-standard') {
+        return MAPS.DEFAULT_XR;
+    }
+
+    const defaultmap = MAPS.DEFAULT;
+    const map = pad.buttons.length < defaultmap.buttons.length ? MAPS.DEFAULT_DUAL : defaultmap;
+    map.mapping = pad.mapping;
+    return map;
+}
+
+let deadZone = 0.25;
+
+/**
+ * @param {number} ms - Number of milliseconds to sleep for.
+ * @returns {Promise<void>}
+ */
+function sleep(ms) {
```

Comment:
I'm assuming this doesn't appear in the API docs? Just wondered if an @ignore was needed...

Reply:
It's not in the class so it should not appear, right?

Reply:
I'm 75% sure it won't appear - worth checking.

Reply:
Ah yeah you were right, it was displayed. That's fixed! Same for getGamepads